File: core/src/main/kotlin/com/alexelkin/pws_library_manager/core/model/Book.kt
package com.alexelkin.pws_library_manager.core.model

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement
import java.util.Locale

@JacksonXmlRootElement(localName = "book")
data class Book(

  @JacksonXmlProperty(isAttribute = true, localName = "version")
  val version: String,

  @JacksonXmlProperty(isAttribute = true, localName = "name")
  val name: String,

  @JacksonXmlProperty(isAttribute = true, localName = "language")
  val language: Locale,

  val displayName: String,

  val displayShortName: String,

  val edition: BookEdition,

  val releaseDate: String,

  val description: String?,

  val preface: String?,

  val creators: List<String>?,

  val editors: List<String>?,

  @JacksonXmlProperty(localName = "psalms")
  val psalmRefs: List<String>
)

File: src/main/scala/v2x/model/Book.scala
package com.alelk.pws.library_manager
package v2x.model

import model.Psalm

import java.util.Locale

case class Book(version: String,
                name: String,
                language: Locale,
                displayName: String,
                displayShortName: String,
                edition: String,
                psalms: List[Psalm],
                releaseDate: Option[String] = None,
                description: Option[String] = None,
                preface: Option[String] = None,
                creators: List[String] = Nil,
                editors: List[String] = Nil)
